### ISSUE TITLE
[CP] [analysis_server] Ensure plugin protocol classes only use types valid for Isolate.send()

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#50650]
### Commit(s) to merge

a7baf9309f682e1ad00464edc9682bcae5d4aceb

### Target

beta

### Prepared changelist for beta/stable

https://dart-review.googlesource.com/c/sdk/+/274200

### Issue Description

The analysis server throws when trying to send messages to plugins because the payloads may contain `HashMap`s which are no longer valid to send across Isolates (see https://github.com/dart-lang/sdk/issues/50594#issuecomment-1333623623 and https://dart-review.googlesource.com/c/sdk/+/273185).

### What is the fix

The fix is to change some explicit uses of `HashMap` which where not required to specifically be `HashMap` to use `Map` literals (`LinkedHashMap`):

```dart
// old
Map<KR, VR> result = HashMap<KR, VR>();
// new
var result = <KR, VR>{};
```

### Why cherry-pick

Without this, analyzer plugins will not work (one of the requests that includes a `HashMap` is one that sets analysis roots, required for basic plugin functionality).

### Risk

low

### Issue link(s)

https://github.com/dart-lang/sdk/issues/50594

### Extra Info

Note: There is a pending CL to list the breaking change that caused this in the changelog at https://dart-review.googlesource.com/c/sdk/+/273185. It has not landed yet and has some outstanding questions, but I believe it's an intended change and won't be reverted (cc @mraleph / @lrhn).

**Comment by itsjustkevin** [comment]
@bwilkerson and @jacob314 thoughts on this cherry-pick?

**Comment by bwilkerson** [comment]
Given that this breaks analyzer plugins, and is a minor change with little risk, I believe that we should cherry-pick it.

**Comment by vsmenon** [comment]
> Given that this breaks analyzer plugins

@bwilkerson - confused by this statement.  Do you mean that a previous change broke plugins and this change fixes that?

**Comment by bwilkerson** [comment]
Exactly.

**Comment by vsmenon** [comment]
Thanks!

lgtm

**Comment by itsjustkevin** [comment]
@DanTup this cherry-pick is approved but the status still shows WIP.  Is this cherry-pick still desired?

**Comment by bwilkerson** [comment]
Yes, the cherry-pick is still desired. Do we just need to send it out for review to get the status changed?

**Comment by DanTup** [comment]
Yep! I wasn't sure what the process was so I only created the CL and didn't request a review. I've requested one now so it doesn't show as WIP now though, but I'm not sure who should review/approve/land it.

Thanks!

**Comment by bwilkerson** [comment]
I went ahead and reviewed it, and it's running on the try bots in case that makes it easier to finish the process.

**Comment by itsjustkevin** [comment]
Thanks a ton @bwilkerson and @DanTup!

**Comment by bwilkerson** [comment]
Given that the CL has been merged, should this issue be closed?

**Comment by itsjustkevin** [comment]
Typically these are not closed until the release is complete.

**Comment by bwilkerson** [comment]
Ok, thanks. Just wanted to make sure this wasn't something that had been missed.

**Comment by itsjustkevin** [comment]
Merged into 2.19.0-444.2.beta.
